### PR TITLE
Link to docs when binary server download fails due to proxy

### LINKED ISSUE #492: Better handle failed  binary downloads due to "407 Proxy Authentication Required"
Telemetry shows a number of users are failing to download the binary because of "Server binary download failed: status code 407".  407 error means Proxy Authentication Required. When such errors are encountered, we could display a popup with (links to) instructions to configure proxy settings for vscode-xml.

### PR DESCRIPTION
**Pull request by datho7561**
Closes #492

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

File: src/server/binary/binaryServerStarter.ts [hunk 1]
```diff
@@ -413,3 +425,7 @@ async function acceptBinaryDownloadResponse(response: http.IncomingMessage): Pro
     response.pipe(serverBinaryFileStream);
   });
 }
+
+async function openProxyDocumentation(): Promise<void> {
```
Comment (fbricon) [inline src/server/binary/binaryServerStarter.ts:429]: can we track the open proxy doc event?

Comment (datho7561) [reply src/server/binary/binaryServerStarter.ts:430]: Okay, I've added a telemetry event for clicking the link. What do you think of the name of the event?

**Comment by fbricon** [comment]
@snjeza I know you frequently test proxy issues, can you please check this one works as expected?

File: src/server/binary/binaryServerStarter.ts (hunk 1 above)
Comment (datho7561) [reply src/server/binary/binaryServerStarter.ts:430]: I forgot to document the event in the USAGE_DATA.md

**Review by rgrunber — COMMENTED** [review]
I set up a basic local proxy using tinyproxy that required authentication and configured it in vscode, leaving out the authentication credentials, since that's what a 407 seems to be about. I saw the popup to configure it correctly as expected.

However, is the telemetry request supposed to be cached for sending at a later time, once the user fixes their proxy settings ? Requests to api.segment.io would get dropped as well without authentication.

**Comment by datho7561** [comment]
It looks like its queued https://github.com/redhat-developer/vscode-redhat-telemetry/blob/main/src/utils/telemetryEventQueue.ts. I'll research if VS Code needs to restart to apply proxy settings.

**Comment by datho7561** [comment]
I added a new page in the documentation that explains how to set up the proxy to work with the extension. It has more specific instructions than the documentation that I was linking to before. Hopefully the fact that its available offline is also helpful.

Its odd that the "download failed" events reach Segment when the proxy is not configured correctly. I guess that would mean whatever proxy is being used allows communication with Segment, but not jboss.org.

File: docs/Proxy.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+VS Code provides a setting called `http.proxySupport`.
+If you set it to `override`, VS Code will attempt to rewrite the requests from extensions to go through the proxy that is configured for VS Code.
+However, this method sometimes does not work correctly.
+
+The tested way to get vscode-xml to use your proxy is to set the following settings:
+ * `http.proxy`
+ * `http.proxyAuthorization`, if you use an authenticated proxy
+
+Once these settings are set,
```
Comment (rgrunber) [inline docs/Proxy.md:22]: Does LemMinX respect the VS Code proxy settings ? I see the JVM has a separate set of properties.

Comment (datho7561) [reply docs/Proxy.md:22]: vscode-xml will check if the user configured JVM args has the parameters to set up the proxy. If they don't, vscode-xml converts the values of these settings into the necessary JVM args and appends them to the existing args.

File: docs/Proxy.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+since the binary language server is downloaded from that website.
+You will also need to make sure that you can access any online XML schemas that you want to work with through the proxy.
+
+## Setting up the proxy in VS Code
+
+VS Code provides a setting called `http.proxySupport`.
+If you set it to `override`, VS Code will attempt to rewrite the requests from extensions to go through the proxy that is configured for VS Code.
+However, this method sometimes does not work correctly.
+
+The tested way to get vscode-xml to use your proxy is to set the following settings:
```
[the remark this answers is not in the document]
Comment (datho7561) [reply docs/Proxy.md:18]: I think it is. I'll add a description for both settings.

File: docs/Proxy.md (hunk 2 above)
Comment (rgrunber) [reply docs/Proxy.md:22]: Right. I can see that in settings/proxySettings.ts.

**Review by rgrunber — APPROVED** [review]
Feel free to merge this. We can address additional issues (if any) in later commits.

**Pull request merged**